pkg/machines/libvirt.es6

```diff
@@ -620,4 +622,86 @@ function doUsagePolling (name, connectionName) {
+    logDebug(`handleEvent(${connectionName}): domain ${name}: got event ${event_}; details: ${info}`);
+
+    // types and details: https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainEventID
+    switch (event_) {
+        case 'lifecycle':
+            let type = info.split(' ')[0];
+            switch (type) {
+                case 'Undefined':
+                    dispatch(undefineVm(connectionName, name));
+                    break;
+
+                case 'Defined':
+                    dispatch(updateOrAddVm({ connectionName, name }));
+                    dispatch(updateVm({connectionName, name, state: 'shut off'}));
+                    break;
+
+                case 'Started':
+                    // transient VMs don't have a separate Defined event, so
+                    // make sure VM exists and read properties on startup
+                    dispatch(updateOrAddVm({ connectionName, name }));
+                    dispatch(updateVm({connectionName, name, state: 'running'}));
+                    spawnVirshReadOnly({connectionName, method: 'dominfo', name})
+                        .then(domInfo => parseDominfo(dispatch, connectionName, name, domInfo));
+                    break;
+
+                case 'Stopped':
+                    dispatch(updateVm({connectionName, name, state: 'shut off', actualTimeInMs: -1}));
+                    // transient VMs don't have a separate Undefined event, so remove them on stop
+                    dispatch(undefineVm(connectionName, name, true));
+                    break;
+
+                default:
+                    logDebug(`Unhandled lifecycle event type ${type} in event: ${line}`);
+            }
+            break;
+
+        case 'metadata-change':
+        case 'device-added':
+        case 'device-removed':
+        case 'disk-change':
+        case 'tray-change':
+        case 'control-error':
+            // these (can) change what we display, so re-read the state
+            dispatch(getVm(connectionName, name));
+            break;
+
+        default:
+            logDebug(`handleEvent ${connectionName} ${name}: ignoring event ${line}`);
+    }
+}
+
+function startEventMonitor(dispatch, connectionName) {
+    var output_buf = '';
+
+    // set up event monitor for that connection; force PTY as otherwise the buffering
+    // will not show every line immediately
+    cockpit.spawn(['virsh'].concat(VMS_CONFIG.Virsh.connections[connectionName].params).concat(['-r', 'event', '--all', '--loop']), {'err': 'message', 'pty': true})
+        .stream(data => {
+            // buffer and line-split the output, there is no guarantee that we always get whole lines
+            output_buf += data;
+            let lines = output_buf.split('\n');
+            while (lines.length > 1)
+                handleEvent(dispatch, connectionName, lines.shift().trim());
+            output_buf = lines[0];
+        })
+        .fail(ex => console.error("Could not start event monitor:", ex));
```

Comment:
`virsh event` will exit when libvirtd is restarted, and maybe for other reasons.  I think we should have a loop here that restarts `virsh event` when it exits.  We probably should wait a while before the restart to avoid a busy loop if virsh never runs properly.  What about measuring the wait period from the start of virsh to the next start (and not from the exit to the next start)? 5 seconds?